### PR TITLE
Feature Create user account from admin interface

### PR DESCRIPTION
**Pull request by R0ckweb**
This allow to create new user account from administration web interface.

Related to #1456 #146 

screen shot 2017-04-11 at 20 06 08

screen shot 2017-04-11 at 20 06 19

screen shot 2017-04-11 at 20 06 47

### CODE REVIEW

**Comment by halkeye** [comment]
Drat you are slightly ahead of me.

someone was reporting earlier that the forgot password functionality was breaking, but didn't look into it yet. As such, I was thiking it might be worth it to be able to specify password. Maybe default to random if not otherwise set.

Are you creating tests for this? Thats what I was getting around to right now.

**Comment by R0ckweb** [comment]
I've been hesitant to add password fields too.
But I think use forgot password function force users to change their password. If  they have a ready to use password I think most of them don't take the time to change it.
Also, according to #146 next step is to implement account password modification from admin ui.

Password fields can easily be added if I'm the only one with this opinion.

Unit test added ;)

File: app/controllers/admin/accounts_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 module Admin
   class AccountsController < BaseController
-    before_action :set_account, except: :index
+    before_action :set_account, except: %i(index new create)
```
Comment (halkeye) [inline app/controllers/admin/accounts_controller.rb]: New and create don't have account IDs in the URL so this function isn't needed to be called

File: spec/controllers/admin/accounts_controller_spec.rb [hunk 2]
```diff
@@ -13,11 +15,104 @@
...
     it 'returns http success' do
-      get :show, params: { id: account.id }
+      get :new
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe 'POST #create' do
+    before do
+      request.env["devise.mapping"] = Devise.mappings[:user]
```
Comment (halkeye) [inline spec/controllers/admin/accounts_controller_spec.rb]: If you are not using password in the form, why is the test providing it?

File: spec/controllers/admin/accounts_controller_spec.rb [hunk 3]
```diff
@@ -13,11 +15,104 @@
...
+      expect(User.find_by(email: 'testadmin@example.com')).to_not be_nil
+    end
+  end
+
+  describe 'POST #create existing user' do
+    let(:user) { Fabricate(:user, email: 'testadmin@example.com', account: Fabricate(:account, username: 'testadmin')) }
+
+    before do
+      request.env["devise.mapping"] = Devise.mappings[:user]
+      post :create, params: { user: { account_attributes: { username: user.account.username }, email: user.email, password: '12345678', password_confirmation: '12345678' } }
```
Comment (halkeye) [inline spec/controllers/admin/accounts_controller_spec.rb]: Same here, why password being provided

**Review by halkeye — CHANGES_REQUESTED** [review]
Minor issues

File: app/controllers/admin/accounts_controller.rb (hunk 1 above)
Comment (R0ckweb) [inline app/controllers/admin/accounts_controller.rb]: This is why they are in 'except' filter !

Comment (halkeye) [inline app/controllers/admin/accounts_controller.rb]: Oh shit. Shouldn't code review from bed without glasses. Sorry sorry

**Comment by halkeye** [comment]
Makes sense if there is eventually going to be a way for admins to set passwords. I withdraw objections.

**Comment by halkeye** [comment]
Maybe it should go to the show page after creation. I know it's kinda rails standard to go-to list though

**Review by halkeye — APPROVED** [review]

**Comment by halkeye** [comment]
@R0ckweb ps, you seem to have merge conflicts now. The code moves so fast.

**Pull request closed without merging**